libbeat/publisher/bc/publisher/client.go

```diff
@@ -240,3 +274,22 @@ func MakeContext(opts []ClientOption) ([]common.MapStr, Context) {
 	}
 	return meta, ctx
 }
+
+func makeEvent(fields common.MapStr, meta common.MapStr) beat.Event {
+	var ts time.Time
+	switch value := fields["@timestamp"].(type) {
+	case time.Time:
```

Comment:
Different places? The bc client is adapting the event, before pushing to the new pipeline. Maybe in some unit tests you mean?